docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,574 @@
+
+This approach requires additional logic if:
+
+* The POCO includes complex properties.
+* You need to handle attributes such as `[JsonIgnore]` or options such as custom encoders.
+
+### Callbacks
+
+`Newtonsoft.Json` lets you execute custom code at several points in the serialization or deserialization process:
+
+* OnDeserializing (when beginning to deserialize an object)
+* OnDeserialized (when finished deserializing an object)
+* OnSerializing (when beginning to serialize an object)
+* OnSerialized (when finished serializing an object)
+
+In <xref:System.Text.Json>, you can simulate callbacks by writing a custom converter. The following example shows a custom converter for a POCO. The converter includes code that displays a message at each point that corresponds to a `Newtonsoft.Json` callback.
+
+[!code-csharp[](~/samples/snippets/core/system-text-json/csharp/WeatherForecastCallbacksConverter.cs)]
+
+Register this custom converter by [using an attribute](system-text-json-converters-how-to.md#registration-sample---jsonconverter-on-a-type) on the class or by [adding the converter](system-text-json-converters-how-to.md#registration-sample---converters-collection) to the <xref:System.Text.Json.JsonSerializerOptions.Converters> collection.
+
+If you use a custom converter that follows the preceding sample:
+
+* The `OnDeserializing` code doesn't have access to the new POCO instance. To manipulate the new POCO instance at the start of deserialization, put that code in the POCO constructor.
+* Don't pass in the options object when recursively calling `Serialize` or `Deserialize`. The options object contains the `Converters` collection. If you pass it in to `Serialize` or `Deserialize`, the converter will be used, making an infinite loop that results in a stack overflow exception.
+
+## Scenarios that JsonSerializer currently doesn't support
+
+Workarounds are possible for the following scenarios, but some of them would be relatively difficult to implement. Code samples for workarounds are not available for these scenarios.
+
+### Types without built-in support
+
+<xref:System.Text.Json> doesn't provide built-in support for the following types:
+
+* <xref:System.Data.DataTable> and related types
+* F# types, such as [discriminated unions](../../fsharp/language-reference/discriminated-unions.md), [record types](../../fsharp/language-reference/records.md), and [anonymous record types](../../fsharp/language-reference/anonymous-records.md).
+* Collection types in the <xref:System.Collections.Specialized> namespace
+* <xref:System.Dynamic.ExpandoObject>
+* <xref:System.TimeZoneInfo>
+* <xref:System.Numerics.BigInteger>
+* <xref:System.TimeSpan>
+* <xref:System.DBNull>
+* <xref:System.Type>
```

Comment:
`ValueTuple<T1>`/`ValueTuple<T1, T2>`/etc. support seems to keep coming up. Add to the list:
https://github.com/dotnet/runtime/issues/1519#issuecomment-572751931